tracing-mock/src/field.rs

```diff
@@ -55,22 +150,108 @@ impl PartialEq for ExpectedValue {
     }
 }
 
-pub fn msg(message: impl fmt::Display) -> ExpectedField {
-    ExpectedField {
-        name: "message".to_string(),
-        value: ExpectedValue::Debug(message.to_string()),
-    }
-}
-
 impl ExpectedField {
-    /// Expect a field with the given name and value.
+    /// Sets the value to expect when matching this field.
+    ///
+    /// If the recorded value for this field is different, the
+    /// expectation will fail.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let event = expect::event()
+    ///     .with_fields(expect::field("field_name").with_value(&"value"));
+    ///
+    /// let (collector, handle) = collector::mock()
+    ///     .event(event)
+    ///     .run_with_handle();
+    ///
+    /// tracing::collect::with_default(collector, || {
+    ///     tracing::info!(field_name = "value");
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
+    ///
+    /// A different value will cause the test to fail:
+    ///
+    /// ```should_panic
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let event = expect::event()
+    ///     .with_fields(expect::field("field_name").with_value(&"value"));
+    ///
+    /// let (collector, handle) = collector::mock()
+    ///     .event(event)
+    ///     .run_with_handle();
+    ///
+    /// tracing::collect::with_default(collector, || {
+    ///     tracing::info!(field_name = "different value");
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
     pub fn with_value(self, value: &dyn Value) -> Self {
         Self {
             value: ExpectedValue::from(value),
             ..self
         }
     }
 
+    /// Adds an additional [`ExpectedField`] to be matched.
+    ///
+    /// Both fields must match, if either of them are not present, or
+    /// if the value for either field is different, the expectation
+    /// will fail.
```

Comment:
```suggestion
    /// _Both_ fields must match for an expectation to pass. If any field is present,
    /// or if the value for either field differs, the expectation will fail. 
    ///
    /// For cases where more than two fields need to be expected, consider
    /// using [`ExpectedFields::only`].
```

Note: should we even have this API? seems a little odd provide two options where one is a generalization on arity.

Reply:
Yeah, that sounds better. I'll make that change. Thanks!